pkg/influx/api.go

```diff
@@ -27,8 +27,15 @@ type API struct {
 	recorder Recorder
 }
 
-func (a *API) Register(server *server.Server, authMiddleware middleware.Interface) {
-	server.HTTP.Handle("/api/v1/push/influx/write", authMiddleware.Wrap(http.HandlerFunc(a.handleSeriesPush)))
+func (a *API) Register(server *server.Server, enableAuth bool) {
+	handler := http.HandlerFunc(a.handleSeriesPush)
+
+	if enableAuth {
+		httpAuthMiddleware := middleware.AuthenticateUser
+		httpAuthMiddleware.Wrap(handler)
```

Comment:
the wrap function returns a wrapped result, so this needs to change to work:
```suggestion
		handler = middleware.AuthenticateUser.Wrap(handler)
```

Reply:
(I think a small unit test **could** be written using InjectOrgIDIntoHTTPRequest -- the wrapper is here: https://github.com/grafana/mimir/blob/main/vendor/github.com/weaveworks/common/middleware/http_auth.go#L10 and makes a call to this code: https://github.com/grafana/mimir/blob/main/vendor/github.com/weaveworks/common/user/http.go#L23.  A minimal artificial httprequest could be created to exercise this code)

Reply:
(I think a small unit test **could** be written using InjectOrgIDIntoHTTPRequest -- the wrapper is here: https://github.com/grafana/mimir/blob/main/vendor/github.com/weaveworks/common/middleware/http_auth.go#L10 and makes a call to this code: https://github.com/grafana/mimir/blob/main/vendor/github.com/weaveworks/common/user/http.go#L23.  A minimal artificial httprequest could be created to exercise this code)

Reply:
yeah that's what I mean -- again not worth totally blocking this if it's > 1/2 day of poking